### PR TITLE
Feature/songs grid Fixes

### PR DESCRIPTION
**Pull request by saumya-mohinani**
### Description
made fixes according to feedback

### CODE REVIEW

File: app/components/AlbumArt/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import styled from 'styled-components';
```
Comment (alichherawalla) [inline app/components/AlbumArt/index.js]: this shouldn't be called AlbumArt anymore, lets give it a generic name

File: app/components/AlbumArt/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import PropTypes from 'prop-types';
+
+import React from 'react';
```
Comment (alichherawalla) [inline app/components/AlbumArt/index.js]: lets not add newlines after every import

File: app/components/AlbumArt/index.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    img.src = highQualitySrc;
+    img.onload = () => {
+      setSrc(highQualitySrc);
+    };
+  }, [lowQualitySrc, highQualitySrc]);
+  return [src, { blur: src === lowQualitySrc }];
+};
+
+const AlbumArt = ({ source }) => {
+  const [src, { blur }] = useProgressiveImg(`https://placeholder.pics/svg/250/DEDEDE/555555/loading...`, source);
```
Comment (alichherawalla) [inline app/components/AlbumArt/index.js]: if we're not gonna use blur, no point destructing the object to get it

File: app/containers/HomeContainer/index.js [hunk 4]
```diff
@@ -55,6 +55,7 @@ export function HomeContainer({
 
   useEffect(() => {
     const loaded = get(reposData, 'items', null) || reposError;
+    console.log(reposError)
```
Comment (alichherawalla) [inline app/containers/HomeContainer/index.js]: remove

File: app/containers/SongContainer/index.js [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import get from 'lodash/get';
+import debounce from 'lodash/debounce';
+import isEmpty from 'lodash/isEmpty';
+import { Card, Skeleton, Input } from 'antd';
+import styled from 'styled-components';
+import { injectIntl } from 'react-intl';
+import { injectSaga } from 'redux-injectors';
+
```
Comment (alichherawalla) [inline app/containers/SongContainer/index.js]: remove extra lines

File: app/containers/SongContainer/index.js [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+  useEffect(() => {
+    const loaded = get(songsData, 'results', null) || songsError;
+    if (loading && loaded) {
+      setLoading(false);
+    }
+  }, [songsData]);
+
+  useEffect(() => {
+    if (query != '' && songsData?.songs?.length) {
```
Comment (alichherawalla) [inline app/containers/SongContainer/index.js:66]: import isEmpty from 'lodash/isEmpty';
...

```if (!isEmpty(query) && songsData?.songs?.length)```

File: app/containers/SongContainer/index.js [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+      setLoading(true);
```
Comment (alichherawalla) [inline app/containers/SongContainer/index.js]: we already have loading in the store, no need for adding another state variable for it. Just use the one in the redux store

File: app/containers/SongContainer/index.js [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+    if (!isEmpty(rName)) {
+      dispatchSongs(rName);
+      setLoading(true);
+    } else {
+      dispatchClearSongs();
+    }
+  };
+  const debouncedHandleOnChange = debounce(handleOnChange, 200);
+
+  //   const refreshPage = () => {
```
Comment (alichherawalla) [inline app/containers/SongContainer/index.js]: remove commented out code

File: app/containers/SongContainer/index.js [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+    return (
+      (items?.length !== 0 || loading) && (
+        <Skeleton loading={loading} active>
+          {/* {totalCount !== 0 && (
+                            <div>
+                                <T id="matching_results" values={{ totalCount }} />
+                            </div>
+                        )} */}
+          <Container>
+            {items?.map((result, index) => (
```
Comment (alichherawalla) [inline app/containers/SongContainer/index.js]: use `For`

File: app/containers/SongContainer/index.js [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+      <T marginBottom={10} id="search_itunes" />
+      <Search
+        data-testid="search-bar"
+        defaultValue={query}
+        type="text"
+        onChange={(evt) => debouncedHandleOnChange(evt.target.value)}
+        onSearch={(searchText) => debouncedHandleOnChange(searchText)}
+      />
+      {renderErrorState()}
+
```
Comment (alichherawalla) [inline app/containers/SongContainer/index.js]: remove new line

File: app/components/AlbumArt/index.js (hunk 1 above)
Comment (saumya-mohinani) [reply app/components/AlbumArt/index.js]: will add a generic name like LazyImage

File: app/components/AlbumArt/index.js (hunk 2 above)
Comment (saumya-mohinani) [reply app/components/AlbumArt/index.js]: got it!

File: app/components/AlbumArt/index.js (hunk 3 above)
Comment (saumya-mohinani) [reply app/components/AlbumArt/index.js]: makes sense, will remove it

File: app/containers/SongContainer/index.js (hunk 7 above)
Comment (saumya-mohinani) [reply app/containers/SongContainer/index.js]: okay , will remove state loading

File: app/containers/SongContainer/index.js (hunk 8 above)
Comment (saumya-mohinani) [reply app/containers/SongContainer/index.js]: sure!

File: app/containers/SongContainer/index.js (hunk 9 above)
Comment (saumya-mohinani) [reply app/containers/SongContainer/index.js]: Okay will use premade components

File: app/containers/HomeContainer/index.js (hunk 4 above)
Comment (saumya-mohinani) [reply app/containers/HomeContainer/index.js]: Sure

File: app/containers/SongContainer/index.js (hunk 6 above)
Comment (saumya-mohinani) [reply app/containers/SongContainer/index.js:66]: Aah, good catch

File: app/components/LazyImage/index.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+import React from 'react';
+
+const ImageArt = styled.img`
+  && {
+    object-fit: contain;
+    height: 250px;
+  }
+`;
+
+const useProgressiveImg = (lowQualitySrc, highQualitySrc) => {
```
Comment (alichherawalla) [inline app/components/LazyImage/index.js:12]: would love to see a custom hook for this. 

You can refer to this implementation: https://github.com/wednesday-solutions/react-screentype-hook

File: app/containers/SongContainer/index.js (hunk 6 above)
Comment (alichherawalla) [reply app/containers/SongContainer/index.js:66]: this needs to be fixed

File: app/containers/SongContainer/index.js [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+    } else {
+      dispatchClearSongs();
+    }
+  };
+  const debouncedHandleOnChange = debounce(handleOnChange, 200);
+
+  const renderResultList = () => {
+    const items = get(songsData, 'results');
+    return (
+      (items?.length !== 0 || loading) && (
```
Comment (alichherawalla) [inline app/containers/SongContainer/index.js:84]:
```
if (items?.length || loading)
```

File: app/components/LazyImage/index.js (hunk 11 above)
Comment (saumya-mohinani) [reply app/components/LazyImage/index.js:12]: okay , will get on it

**Pull request merged**